lms/product/plugin/course_copy.py

```diff
@@ -0,0 +1,74 @@
+from lms.models import File
+from lms.services.exceptions import CanvasFileNotFoundInCourse
+
+
+class CourseCopyPlugin:
```

Comment:
Base CourseCopy plugin

Reply:
Plugins are services that are implemented once per product.

I like the idea of a CourseCopy oriented one because it give more visibility to that problem/functionality. Otherwise is just a implantation detail in each of the product's APIs.